dataprofiler/data_readers/csv_data.py

```diff
@@ -104,23 +104,163 @@ def _check_and_return_options(self, options):
+        
+            # Reverse to start at bottom
+            # Fewer columns are okay, as long as earlier in file
+            proposed_dataset.reverse()
+            for proposed_row in proposed_dataset:
+
+                # Ensure there's a prior_col_count
+                if not prior_col_count:
+                    prior_col_count = len(proposed_row)
+
+                # Ensure's it's not a description row, then ensures cells consistent
+                if len(proposed_row) != prior_col_count and len(proposed_row) > 1:
+                    incorrect_delimiter_flag = True
+                    break
+
+
+                # Evalutes each column for potential issues
+                # e.g. '3' being a seperator, while a single row of integers
+                cell_content = []
+                cell_types = []
+                prior_cell_type = None
+                for col_id in range(len(proposed_row)):
+                    proposed_cell = proposed_row[col_id]
+
+                    col_types[col_id] = proposed_cell['type']
+
+                    # Handle if alpha character are seperator                     
+                    # NOTE: delimiter needs two ajoining cells to flag
+                    if proposed_cell['type'] in ['str', 'none'] \
+                       and prior_cell_type in ['str', 'none']:
+                        if proposed_delim.isalpha():
+                            cell_type_safe_flag = False
+                            
+                        # Ensures when spaces occur, there are 3 or more cols
+                        if proposed_delim == ' ' and 2 >= len(proposed_row):
+                            cell_type_safe_flag = False
+
+                    # Handle if integer characters are seperators
+                    # NOTE: delimiter need one adjoining cell to flag
+                    if proposed_delim_type == 'int' and proposed_cell['type']=='int':
+                        cell_type_safe_flag = False
+
+                    # Handle if start or end is always none
+                    if proposed_cell['type']!='none':
+                        if col_id == len(proposed_row)-1:
+                            row_end_safe_flag = True
+                        if col_id == 0:
+                            row_start_safe_flag = True
+                    
+                    prior_cell_type = proposed_cell['type']
+
+            edge_safe_flag = row_end_safe_flag or row_start_safe_flag
+
+            # Add a delimiter candidate safe to do so
+            if not incorrect_delimiter_flag and cell_type_safe_flag and edge_safe_flag:
+                validated_proposed_delimiters[proposed_delim] = prior_col_count
+                
+                # Find one with most columns 
+                if not delimiter:
+                    delimiter = proposed_delim
+                if prior_col_count \
+                   and prior_col_count > validated_proposed_delimiters[delimiter]:
```

Comment:
Why is the highest column count the basis for which delimiter we use. Couldnt there be a case where that isnt always true?